### PR TITLE
Login

### LINKED ISSUE #13: Users Register and Sing-in
- Register Users:
  - The app must include a register option for users in the menu on the home page
  - [x] There must be a button option for registering a new user
  - [x] The user must be prompted to include his name
  - [x]  The user must give an email
  - [x]  The user must give a password
  - [x]  The user must confirm the password

- Users Sing-in:
  - The app must let users sign in with their email and password
  - [x] The user must give his email and his password to be granted access

### PR DESCRIPTION
**Pull request by Ballo1988**
Added new user register and login, only a registered user can create and edit events, but everyone can see the event list, also added a new button so users can log and register to the website.
More details here #13 
image
image
image
image

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "[shellscript]": {
+    "files.eol": "\n"
+  }
```
Comment (juperameza) [inline .vscode/settings.json:4]: I believe this shouldn't be here

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -38,3 +22,12 @@ body {
...
   background-color: hsl(219, 100%, 30%);
 }
+
+.navbar-image img{
+  width: 400px !important;
+  height: auto !important;
+}
+.navbar-brand {
+  width: 400px !important; /* Adjust the width as needed */
+  height: auto !important; /* Maintain the aspect ratio */
```
Comment (juperameza) [inline app/assets/stylesheets/application.scss]: Remember this isn't a good practice why are you using it?

File: app/controllers/events_controller.rb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 class EventsController < ApplicationController
   before_action :set_event, only: %i[update destroy edit]
+  before_action :authenticate_user!, only: %i[new create edit update destroy]
```
Comment (juperameza) [inline app/controllers/events_controller.rb]: If you're gonna use it always, just do `beforr_action : authenticate_user!`

File: app/controllers/events_controller.rb [hunk 4]
```diff
@@ -40,13 +38,18 @@ def destroy
     redirect_to root_path, status: :see_other
   end
 
+  def log_out
+    sign_out(current_user)
+    redirect_to root_path
+  end
+
```
Comment (juperameza) [inline app/controllers/events_controller.rb]: Why is this here??

File: config/routes.rb [hunk 5]
```diff
@@ -1,7 +1,13 @@
 Rails.application.routes.draw do
-  root "events#index"
+  devise_for :users
 
-  get '/events/show', to: 'events#show', as: 'show_event'
+  # Add the log out route
+  devise_scope :user do
+    get '/log_out', to: 'devise/sessions#destroy', as: 'log_out'
```
Comment (juperameza) [inline config/routes.rb]: Again why are you doing this

**Review by juperameza — CHANGES_REQUESTED** [review]
If you were just gonna use one devise view you should've specific it when generating it `rails generate devise:views -v registrations confirmations`
And try to add that other controller with that index action do you follow all the best practices

File: app/controllers/events_controller.rb (hunk 3 above)
Comment (hugocruz1234) [reply app/controllers/events_controller.rb]: users must be logged in to view public events or they can view these events without being logged in?

Comment (juperameza) [reply app/controllers/events_controller.rb]: But you're letting them see all events without beings logged in 
My first comment was wrong you're omitting the show action but that means every user can see all the events without being logged in

Comment (hugocruz1234) [reply app/controllers/events_controller.rb]: yes, that's why I asked the previous ones, to see if we leave it as before, that anyone can see public events without session or if the session is necessary to be able to see those events.

File: app/controllers/events_controller.rb (hunk 4 above)
Comment (Ballo1988) [reply app/controllers/events_controller.rb]: We don't quite understand, Should we move it to the Application controller?

Comment (juperameza) [reply app/controllers/events_controller.rb]: Devise already has a logout route

Comment (juperameza) [reply app/controllers/events_controller.rb]: You should use it instead of creating your own

Comment (juperameza) [reply app/controllers/events_controller.rb]: You should use it instead of creating your own

Comment (Ballo1988) [reply app/controllers/events_controller.rb]: Got it, thanks!

**Review by hugocruz1234 — APPROVED** [review]

**Review by hugocruz1234 — APPROVED** [review]

File: config/routes.rb [hunk 6]
```diff
@@ -1,7 +1,10 @@
 Rails.application.routes.draw do
-  root "events#index"
+  devise_for :users
 
-  get '/events/show', to: 'events#show', as: 'show_event'
+  # Add the log out route
+  devise_scope :user do
+  end
```
Comment (juperameza) [inline config/routes.rb]:
```suggestion
```

**Review by juperameza — CHANGES_REQUESTED** [review]

**Review by juperameza — APPROVED** [review]
LGTM!
